### PR TITLE
Windows master

### PR DESCRIPTION
**Pull request by dparnell**
There are quite a few bugs that have been fixed in the upstream master that are still present in the windows branch. I've done a merge of master into the windows branch. It is almost working (at least it compiles). What I'm seeing is the app locks up if the config specifies that the window should have decorations. If I turn off decorations then it works but of course moving and resizing the window no longer work. Not being particularly familiar with rust I'm hoping you may be able to see what's going on. I'm looking at it here also.

### CODE REVIEW

**Comment by dparnell** [comment]
ok, now it is starting. However I am still seeing corruption when I run emacs inside a tmux session over ssh :cry:

**Comment by zacps** [comment]
I had been planning to wait until all the issues were sorted out with the windows build before rebasing on master, however it'd be a shame to throw away all of the work you did on the merge.
I'll take a look at it, did you see the corruption in the current windows branch?

**Comment by dparnell** [comment]
I only see the corruption on windows. On a mac it seems to render as expected.

**Comment by zacps** [comment]
Sorry, I meant on my current windows branch; as in, did the rebase cause the issue or was it something that should have been fixed by a commit in the rebase.

**Comment by dparnell** [comment]
I was hoping it would be something that master should have fixed. I'm seeing the same behaviour on all windows branches that I've tried. There was a commit on master that sounded like it should fix a problem similar to this, but I guess there is something else wrong.

File: src/window.rs [hunk 1]
```diff
@@ -177,19 +181,31 @@ impl From<glutin::ContextError> for Error {
...
+    pub fn new(
+        title: &str,
+        window_config: &WindowConfig,
+    ) -> Result<Window> {
         let event_loop = EventsLoop::new();
 
+        #[cfg(not(windows))]
+        let initially_shown = false;
+        #[cfg(windows)]
+        let initially_shown = true;
```
Comment (zacps) [inline src/window.rs:194]: What commit is this from? I heard something about this being an issue on OSX, why is it just windows that has a window that's initially shown?

**Comment by zacps** [comment]
Other than that the changes look good to me. Could you give me a little more detail on the corruption issue? Screenshots, steps to reproduce, etc.

**Comment by dparnell** [comment]
To reproduce this:



1. `ssh user@linuxbox`
2. `tmux`
3. Control-B % - split horizontally
4. `htop`

File: src/window.rs (hunk 1 above)
Comment (dparnell) [reply src/window.rs:194]: That was some code I wrote to stop the app from locking up at the start. I've not had any problems running master under MacOS

**Comment by zacps** [comment]
I got some weird artifacts from resizing but nothing that I haven't seen before from an SSH connection.

As this isn't a regression I'm going to merge this. If it happens consistently feel free to open an issue, though it might be hard for me to diagnose.
My guess is it's an issue with winpty, if anything because winpty manages the buffer and alacritty just forwards events to it.

**Comment by dparnell** [comment]
Yeah, I'm guessing it is something related to winpty. It's almost like it is somehow not correctly handling scrolling regions. I suspect that if I split the tmux session the other way I will not see the same issues.

**Comment by dparnell** [comment]
Interestingly I am not seeing as many issues if I use `screen` rather than `tmux`

**Comment by dparnell** [comment]
It seems to be behaving in the same way as this bug: https://github.com/jwilm/alacritty/pull/747

**Comment by dparnell** [comment]
I just thought I'd try something and I ran up a standard Windows cmd window and used that to run ssh to connect to my Linux machine and I'm seeing exactly the same behavior as in Alacritty. This means the problem is somewhere else. Maybe the ssh client I'm using is rubbish. I'll look around and see what I can see.

**Comment by dparnell** [comment]
I have installed the official Windows 10 SSH client and it seems to behave properly. I was using the Cygwin SSH client and it looks like this was fighting with WinPTY. Now to get the Alt key working properly so I can run Emacs 😄

**Pull request merged**